services/QuillLMS/app/controllers/api/v1/session_feedback_histories_controller.rb

```diff
@@ -16,17 +16,11 @@ def index
     }
   end
 
-  def session_data_for_csv
+  def email_csv_data
     options = params.permit(:activity_id, :start_date, :end_date, :filter_type, :responses_for_scoring).to_h.symbolize_keys
-    feedback_histories = FeedbackHistory.session_data_for_csv(**options)
-    results = []
-    feedback_histories.find_each(batch_size: 10_000) { |feedback_history| results << feedback_history.serialize_csv_data }
-    results.sort! { |a,b| b["datetime"] <=> a["datetime"] }
-    if results
-      render json: results
-    else
-      render plain: "The resource you were looking for does not exist", status: 404
-    end
+    email = current_user.email
+    InternalTool::EmailFeedbackHistorySessionDataWorker.perform_async(options[:activity_id], options[:start_date], options[:end_date], options[:filter_type], options[:responses_for_scoring], email)
```

Comment:
For readability, I like to break method calls with 3+ parameters into multiple lines:
```
  InternalTool::EmailFeedbackHistorySessionDataWorker.perform_async(
    options[:activity_id], 
    options[:start_date], 
    options[:end_date], 
    options[:filter_type], 
    options[:responses_for_scoring], 
    current_user.email
  )
```

Also, no need for the temp variable `email` as it's only used as a parameter